### PR TITLE
Rss download development branch

### PR DESCRIPTION
**Pull request by Julesssss**
Testing pull requests

### CODE REVIEW

**Comment by amitav13** [comment]
We need to update the .gitignore, should get rid of all *.iml files and the idea folder. Will do that in the next PR.

File: app/build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@ android {
 
     defaultConfig {
         applicationId "infrmr.newsapp.github.com.ifrmr"
-        minSdkVersion 19
+        minSdkVersion 15
```
Comment (amitav13) [inline app/build.gradle:9]: We should start out trying to support 8 and up. We can move the min version up if we come across a dependency that needs a certain sdk, but the goal should be to support as many devices as possible.

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/MainActivity.java [hunk 2]
```diff
@@ -31,9 +150,157 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+            myWebView.loadData(result, "text/html", null);
+        }
+    }
+
+
+    // Uploads XML from stackoverflow.com, parses it, and combines it with
+    // HTML markup. Returns HTML string.
+    private String loadXmlFromNetwork(String urlString) throws XmlPullParserException, IOException {
+        InputStream stream = null;
+        StackOverflowXmlParser stackOverflowXmlParser = new StackOverflowXmlParser();
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/MainActivity.java]: This line is giving me an error. Is it in a local file of yours? This needs to be fixed before merging!

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/SettingsActivity.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import android.os.Bundle;
+import android.preference.PreferenceActivity;
+
+/**
+ * Created by user on 22-Apr-15.
+ */
+
+
+
+public class SettingsActivity extends PreferenceActivity implements SharedPreferences.OnSharedPreferenceChangeListener {
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/SettingsActivity.java:13]: Preference activity is deprecated...try using Preference Fragment instead.

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/TheVergeXmlParser.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        // todo published
+        String content = null;
+        while (parser.next() != XmlPullParser.END_TAG) {
+
+            if (parser.getEventType() != XmlPullParser.START_TAG) {
+                continue;
+            }
+
+            String name = parser.getName();
+            if (name.equals("title")) {
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/TheVergeXmlParser.java]: Could just use a switch here

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/TheVergeXmlParser.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    // Processes link tags in the feed.
+    private String readLink(XmlPullParser parser) throws IOException, XmlPullParserException {
+        String link = "";
+        parser.require(XmlPullParser.START_TAG, ns, "link");
+        String tag = parser.getName();
+        String relType = parser.getAttributeValue(null, "rel");
+        if (tag.equals("link")) {
+            if (relType.equals("alternate")) {
+                link = parser.getAttributeValue(null, "href");
+                Log.d(getClass().getSimpleName(), "link: " + link.toString());
```
Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/TheVergeXmlParser.java]: link.toString is redundant

Comment (holysheep) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/TheVergeXmlParser.java]: that's true
going to be simplified

**Comment by amitav13** [comment]
Also, we should refactor things out of MainActivity into a fragment. Think very few activities, lots of fragments!

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/MainActivity.java (hunk 2 above)
Comment (Julesssss) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/MainActivity.java]: Which line?

Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/MainActivity.java]: 192.

Comment (amitav13) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/MainActivity.java]: StackOverflowXmlParser isn't defined anywhere

Comment (Julesssss) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/MainActivity.java]: Its the name of the entire Class. I forgot to change it to 'VergeXmlParse.class'

File: app/src/main/java/infrmr/newsapp/github/com/ifrmr/SettingsActivity.java (hunk 3 above)
Comment (Julesssss) [inline app/src/main/java/infrmr/newsapp/github/com/ifrmr/SettingsActivity.java:13]: Done

**Pull request merged**